python/lsst/cp/verify/mergeResults.py

```diff
@@ -54,19 +73,61 @@ class CpVerifyExpMergeConnections(pipeBase.PipelineTaskConnections,
         storageClass="StructuredDataDict",
         dimensions=["instrument", "exposure"],
     )
+    outputResults = cT.Output(
+        name="exposureResults",
+        doc="Output results.",
+        storageClass="ArrowAstropy",
+        dimensions=["instrument", "exposure"],
+    )
+    outputMatrix = cT.Output(
+        name="exposureMatrix",
+        doc="Output matrix.",
+        storageClass="ArrowAstropy",
+        dimensions=["instrument", "exposure"],
+    )
+
+    def __init__(self, *, config=None):
+        super().__init__(config=config)
+
+        if not self.config.hasMatrixCatalog:
+            self.inputs.remove("inputMatrix")
+            self.outputs.remove("outputMatrix")
+        if not self.config.hasInputResults:
+            self.inputs.remove("inputResults")
 
 
 class CpVerifyExpMergeConfig(pipeBase.PipelineTaskConfig,
                              pipelineConnections=CpVerifyExpMergeConnections):
     """Configuration parameters for exposure stats merging.
     """
-    exposureStatKeywords = pexConfig.DictField(
+    statKeywords = pexConfig.DictField(
         keytype=str,
         itemtype=str,
         doc="Dictionary of statistics to run on the set of detector values. The key should be the test "
         "name to record in the output, and the value should be the `lsst.afw.math` statistic name string.",
         default={},
     )
+    hasMatrixCatalog = pexConfig.Field(
+        dtype=bool,
+        doc="Is there matrix catalog to merge?",
+        default=False,
+    )
+    hasInputResults = pexConfig.Field(
+        dtype=bool,
+        doc="Are there results tables to merge?",
+        default=False,
+    )
+
+    thisDimension = pexConfig.Field(
+        dtype=str,
+        doc="Dimension name for this input.",
+        default="detector",
+    )
```

Comment:
So this works because the quantum dimensions (default are `"instrument", "exposure"`) must not include whatever is listed here?  If this is the case, there should be a check in verify that `thisDimension` is not listed above.  
At the same time, I don't understand the name (`thisDimension`) or the docstring (`"Dimension name for this input."`).

Reply:
The `thisDimension` field is intended to be the dimension that we're merging over.  This allows the same class and code to be able to merge over a set of inputs with individual dimensions `{instrument, exposure, detector}` over `detector` and then to merge the outputs of that step (which now have dimensions `{instrument, exposure}`) over `exposure`.

Reply:
If it's the dimension to merge over, may I suggest a name of `mergeDimension`?